x/bridge/observer/observer.go

```diff
@@ -116,31 +139,69 @@ func (o *Observer) sendOutbound(ctx context.Context) {
 
 	for srcId, queue := range o.outTxQueue {
 		srcChain := o.chains[srcId]
-		height, err := srcChain.Height()
+		height, err := srcChain.Height(ctx)
 		if err != nil {
 			o.logger.Error(fmt.Sprintf("Failed to get height for %s: %s", srcId, err.Error()))
 			continue
 		}
-		confirmationsRequired, err := srcChain.ConfirmationsRequired()
-		if err != nil {
-			o.logger.Error(fmt.Sprintf(
-				"Failed to get confirmations required for %s: %s",
-				srcId,
-				err.Error(),
-			))
-			continue
-		}
-		var newQueue []Transfer
+
+		var newQueue []TxQueueItem
 		for _, out := range queue {
-			if height-out.Height < confirmationsRequired {
+			confirmed, err := o.isTxConfirmed(ctx, height, &out)
+			if !confirmed || err != nil {
+				if err != nil {
+					o.logger.Error(fmt.Sprintf("Failed to confirm tx %s", err.Error()))
+				}
 				newQueue = append(newQueue, out)
 				continue
 			}
-			err = o.chains[out.DstChain].SignalInboundTransfer(ctx, out)
+			err = o.chains[out.Tx.DstChain].SignalInboundTransfer(ctx, out.Tx)
```

Comment:
I would prefer just checking error first, then checking !confirmed, the nested if is less understandable